Hometask8/src/test/java/homework23/api/BuyBookAPITest.java

```diff
@@ -0,0 +1,138 @@
+package homework23.api;
+
+import homework23.api.dto.CartResponseDTO;
+import homework23.api.dto.OrderDTO;
+import homework23.api.dto.UserResponseDTO;
+import homework23.api.dto.UserDTO;
+import homework23.api.services.UsersService;
+import io.restassured.RestAssured;
+import io.restassured.parsing.Parser;
+import io.restassured.response.Response;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static org.testng.Assert.assertEquals;
+
+public class BuyBookAPITest {
+    UserDTO defaultUser;
+    String token;
+    String userId;
+    int bookId = 2;
+    UsersService usersService = new UsersService();
+
+    @Test
+    public void createNewUserTest() {
+        RestAssured.defaultParser = Parser.JSON;
+        defaultUser = TestData.buildDefaultUserData();
+
+        Response response = usersService.createNewUser(defaultUser);
+
+        response.then().statusCode(200).extract().asString();
+    }
+
+    @Test
+    public void createNewUserWithoutRequiredDataEntryTest() {
+        RestAssured.defaultParser = Parser.JSON;
+        UserDTO expectedUser = TestData.buildUserDataWithoutPasswordField();
+
+        Response response = usersService.createNewUser(expectedUser);
+
+        response.then().statusCode(500).extract().asString();
+    }
+
+    @Test
+    public void createNewUserWithWrongFieldTypeTest() {
+        RestAssured.defaultParser = Parser.JSON;
+        UserDTO expectedUser = TestData.buildUserDataWithWrongFieldType();
+
+        Response response = usersService.createNewUser(expectedUser);
+
+        response.then().statusCode(500).extract().asString();
+    }
+
+    @Test
+    public void loginUserTest() {
+        RestAssured.defaultParser = Parser.JSON;
+        createNewUserTest();
```

Comment:
Instead of calling one test from another, you should refactor the common setup code into a separate method. This method can then be called from each test that needs it. This maintains test independence while avoiding code duplication.
Or at least you can mark current method as dependent on createNewUserTest() and do not call it inside current method.
The same for other similar cases.